### ISSUE TITLE
HitSearcher is fetching data from Algolia already when just initially adding a stream listener

### ISSUE THREAD

**Issue by derFunk** [algolia/algoliasearch-helper-flutter#122]
### **Describe the bug 🐛**  

**TL;DR:**
Algolia backend gets called immediately after adding a listener function to the response stream, even if not even actively starting a search by executing the `hitsSearcher.query(String query)` method.

**Long version:**

I'm following [Getting started with Flutter Helper](https://www.algolia.com/doc/guides/building-search-ui/getting-started/flutter/), but want to use `algolia_helper_flutter` without `infinite_scroll_pagination`  and following clean architecture using the BLoC pattern.

I'm initializing the HitsSearcher in my BLoC constructor the common way like this:
```dart
_productsSearcher = HitsSearcher(applicationID: 'latency',
                                        apiKey: '927c3fe76d4b52c5a2912973f35a3077',
                                        indexName: 'STAGING_native_ecom_demo_products');
```

I also want to add a listener to the responses stream as soon as responses are coming in after queries.

I could do that with
```dart
_productsSearcher.responses.listen(_searchPageListener);
```

or, like in the documentation, with
```dart
_searchPage.listen(_searchPageListener);
```

where _searchPage is a getter returing the HitsPage stream:
```dart
  Stream<HitsPage> get _searchPage {
    return _happeningsSearcher.responses.map(HitsPage.fromResponse);
  }
```

⚠️ **As soon as I call `listen()` to add the listener function, HitsSearcher executes a call to the Algolia backend. This I don't want, because I'm still in the setup phase in my BLoC constructor.**

This call is happening because when accessing the responses stream, this code is executed: `searchService.search(req.state)`:
https://github.com/algolia/algoliasearch-helper-flutter/blob/866fbf853c1e47d93f2b1238587490b039c0aedf/helper/lib/src/searcher/hits_searcher.dart#L272

..which agains executes this:
https://github.com/algolia/algoliasearch-helper-flutter/blob/866fbf853c1e47d93f2b1238587490b039c0aedf/helper/lib/src/service/algolia_hits_search_service.dart#L77

### **To Reproduce 🔍**  

1. Initialize HitsSearcher()
2. Add a listener function to the response stream
3. :x: See listener function being called immediately, even without a call to hitsSearcher.query() before.

### **Expected behavior 💭**  
Algolia search does not get called immediately when adding the response listener, but only after calling searcher.query()

### **Environment:**

- OS: Android API 33
- Library Versions
  - algolia_helper_flutter: 0.3.0
  - algoliasearch: 0.3.0
  - algolia_client_core: 0.3.0
  - algolia_client_search: 0.3.0

### **Additional context**  
Thanks for supporting Flutter ❤️

**Comment by ivangr1** [comment]
I am encountering the same issue. Did you manage to solve it maybe?

**Comment by apetroaeiandrei** [comment]
One Year Later... How did you work around this?
